src/SwedbankPay.Sdk.Infrastructure/PaymentInstruments/Swish/SaleListItemDto.cs

```diff
@@ -2,14 +2,25 @@
 
 namespace SwedbankPay.Sdk.PaymentInstruments.Swish
 {
-    internal class SaleListItemDto
-    {
-        public DateTime Date { get; }
-        public Uri Id { get; }
-        public string PayerAlias { get; }
-        public string PaymentRequestToken { get; }
-        public string SwishPaymentReference { get; }
-        public string SwishStatus { get; }
-        public TransactionDto Transaction { get; }
-    }
+	internal class SaleListItemDto
+	{
+		public SaleListItemDto(DateTime date, Uri id, string payerAlias, string paymentRequestToken, string swishPaymentReference, string swishStatus, TransactionDto transaction)
+		{
+			Date = date;
+			Id = id;
+			PayerAlias = payerAlias;
+			PaymentRequestToken = paymentRequestToken;
+			SwishPaymentReference = swishPaymentReference;
+			SwishStatus = swishStatus;
+			Transaction = transaction;
+		}
+
+		public DateTime Date { get; }
+		public Uri Id { get; }
+		public string PayerAlias { get; }
+		public string PaymentRequestToken { get; }
+		public string SwishPaymentReference { get; }
+		public string SwishStatus { get; }
+		public TransactionDto Transaction { get; }

```

Comment:
Since I assume this DTO is used for deserialization, is it not better to just make the properties have public getters instead of adding a constructor? DTO objects are only for data transfer, after all.

Reply:
public setters you mean?

Reply:
Yes, sorry.